### PR TITLE
Rename default module

### PR DESCRIPTION
**Pull request by SuperQ**
To avoid confusion that `default` is special, rename it to `if_mib`
based on the primary MIB.
* Use `ifIndex` for default interface indexing instead of `ifDescr`.
* Make `ifDescr` its own module.
* Add `ifAlias` module.
* Update docs.
* Update `snmp.yml`.

### CODE REVIEW

**Comment by brian-brazil** [comment]
I'm fine with the general thurst of this PR as we've previously discussed.

I'm not so sure about changing the name from `default` to `interfaces`, there's a lot of docs/references to that already.

We'll also need to do something about existing dashboards which presume the current default, at least the example console template (which we should probably just delete).

**Comment by SuperQ** [comment]
I wanted to change the name, as I've had many cases where people confuse `default` with being a global defaults.

File: generator/README.md [hunk 1]
```diff
@@ -41,14 +41,29 @@ Put the extracted mibs in a location NetSNMP can read them from. `$HOME/.snmp/mi
 
 https://github.com/librenms/librenms/tree/master/mibs can also be a good source of MIBs.
 
-
+## Generator file format
```
Comment (brian-brazil) [inline generator/README.md]: This needs work generally, I'd leave it to another PR. For example all the auth stuff needs to move over here.

**Comment by SuperQ** [comment]
@RichiH @brian-brazil PTAL

File: generator/generator.yml [hunk 2]
```diff
@@ -1,20 +1,20 @@
 modules:
-# If ifDesc is unique.
-  default:
+  # Default IF-MIB interfaces table with ifIndex.
+  if-mib:
```
Comment (brian-brazil) [inline generator/generator.yml]: as discussed, let's keep with underscores for consistency.

File: generator/generator.yml [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
+  # Interfaces if ifName is unique.
+  if-mib_ifname:
     walk: [sysUpTime, interfaces, ifXTable]
     lookups:
       - old_index: ifIndex
         # Use OID to avoid conflict with Netscaler NS-ROOT-MIB.
         new_index: 1.3.6.1.2.1.31.1.1.1.1 # ifName
-# If only ifIndex is unique.
-  default_ifindex:
+  # Interfaces if ifDescr is unique.
```
Comment (brian-brazil) [inline generator/generator.yml]: We should add ifAlias too for completelness

**Comment by SuperQ** [comment]
@brian-brazil Finished cleanup, added `snmp.yml` update.
